client/clip_client/client.py

```diff
@@ -284,9 +289,14 @@ async def aencode(self, content, **kwargs):
         )
 
         results = DocumentArray()
-        async for da in self._async_client.post(
-            **self._get_post_payload(content, kwargs)
-        ):
+
+        parameters = kwargs.get('parameters', {})
+        model_name = parameters.get('model', '')
+        payload = self._get_post_parameters(content, kwargs)
+        payload.update(on=f'/encode/{model_name}'.rstrip('/'))
+        payload.update(inputs=self._iter_doc(content))
+
```

Comment:
The `update` is not needed any more. You can directly pass the parameters `on=...` and `inputs` in `post`